src/components/Tokens/TokenTable/TokenTable.tsx

```diff
@@ -43,41 +61,60 @@ const NoTokenDisplay = styled.div`
   padding: 0px 28px;
   gap: 8px;
 `
-const TokenRowsContainer = styled.div`
-  width: 100%;
-`
 
-function useFilteredTokens(tokens: TokenData[] | undefined) {
-  const filterString = useAtomValue(filterStringAtom)
+const DEFAULT_FILTER_STRING = ''
+const DEFAULT_SHOW_FAVORITES = false
+
+const hasNonDefaultFilters = (filterString: string, showFavorites: boolean) =>
+  filterString !== DEFAULT_FILTER_STRING || showFavorites !== DEFAULT_SHOW_FAVORITES
+
+function useFilteredTokens(
+  tokenData: TokenData[],
+  filterString: string,
+  showFavorites: boolean,
+  nonDefaultFiltersExist: boolean
+) {
   const favoriteTokenAddresses = useAtomValue(favoritesAtom)
-  const showFavorites = useAtomValue(showFavoritesAtom)
   const shownTokens =
```

Comment:
i think you'll need to memo this for the below memo to work properly because it's a dep of that memo. it does say it creates a shallow copy of a portion of the array (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter), but i believe the array itself will be a new ref

Reply:
+1